thrust/thrust/detail/inner_product.inl

```diff
@@ -50,17 +58,21 @@ _CCCL_EXEC_CHECK_DISABLE
 template <typename DerivedPolicy,
           typename InputIterator1,
           typename InputIterator2,
-          typename OutputType,
+          typename InitType,
           typename BinaryFunction1,
           typename BinaryFunction2>
-_CCCL_HOST_DEVICE OutputType inner_product(
-  const thrust::detail::execution_policy_base<DerivedPolicy>& exec,
-  InputIterator1 first1,
-  InputIterator1 last1,
-  InputIterator2 first2,
-  OutputType init,
-  BinaryFunction1 binary_op1,
-  BinaryFunction2 binary_op2)
+_CCCL_HOST_DEVICE ::cuda::std::__accumulator_t<BinaryFunction1,
+                                               ::cuda::std::__accumulator_t<BinaryFunction2,
+                                                                            ::cuda::std::iter_value_t<InputIterator1>,
+                                                                            ::cuda::std::iter_value_t<InputIterator2>>,
```

Comment:
We should have an alias for that, where we take iterators and use `iter_value_t` 

How about `__iter_accumulator_t`?